akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -38,14 +37,29 @@ abstract class RouteDirectives extends RespondWithDirectives {
   /**
    * Java-specific call added so you can chain together multiple alternate routes using comma,
    * rather than having to explicitly call route1.orElse(route2).orElse(route3).
+   * @deprecated Use the `RouteDirectives.routes` method instead.
    */
+  @Deprecated
   @CorrespondsTo("concat")
   @varargs def route(alternatives: Route*): Route = RouteAdapter {
     import akka.http.scaladsl.server.Directives._
 
+    require(alternatives.nonEmpty, "Chaining empty list of routes is illegal.")
```

Comment:
please as `if (alternatives.isEmpty) throw ...` since require() allocates a closure and is used quite a bit nowadays so we don't want to add too much overhead there; even though we're deprecating it